### PR TITLE
Add AWS MSK cluster resource

### LINKED ISSUE #6653: Feature Request: Managed Streaming for Kafka
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS now offers managed Kafka clusters

### New or Affected Resource(s)

* aws_msk_cluster

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* https://docs.aws.amazon.com/msk/index.html#lang/en_us

### LINKED ISSUE #8429: Amazon MSK tagging
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

As of [AWS SDK v1.19.14](https://github.com/aws/aws-sdk-go/releases/tag/v1.19.14), merged in https://github.com/terraform-providers/terraform-provider-aws/pull/8381, AWS MSK Kafka clusters support resource tagging.

### PR DESCRIPTION
**Pull request by jesseaukeman**
This is a new implementation of the AWS MSK cluster resource.

I realize this overlaps and duplicates #8357.   This was a separate and parallel effort undertaken for the company I work for.  I apologize if this is bad form, however I think this effort is a bit further along, as the other effort may be stalled on acceptance tests.

* implement aws_msk_cluster resource with support for tags
* implement acceptance tests

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Fixes #6653 
Fixes #8429

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note

```

Output from acceptance testing:

```
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSMskCluster'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -parallel 20 -run=TestAccAWSMskCluster -timeout 120m
=== RUN   TestAccAWSMskCluster_basic
=== PAUSE TestAccAWSMskCluster_basic
=== RUN   TestAccAWSMskCluster_kms
=== PAUSE TestAccAWSMskCluster_kms
=== RUN   TestAccAWSMskCluster_enhancedMonitoring
=== PAUSE TestAccAWSMskCluster_enhancedMonitoring
=== RUN   TestAccAWSMskCluster_tagsUpdate
=== PAUSE TestAccAWSMskCluster_tagsUpdate
=== RUN   TestAccAWSMskCluster_brokerNodes
=== PAUSE TestAccAWSMskCluster_brokerNodes
=== CONT  TestAccAWSMskCluster_basic
=== CONT  TestAccAWSMskCluster_tagsUpdate
=== CONT  TestAccAWSMskCluster_kms
=== CONT  TestAccAWSMskCluster_brokerNodes
=== CONT  TestAccAWSMskCluster_enhancedMonitoring
--- PASS: TestAccAWSMskCluster_enhancedMonitoring (958.48s)
--- PASS: TestAccAWSMskCluster_kms (1062.45s)
--- PASS: TestAccAWSMskCluster_basic (1063.50s)
--- PASS: TestAccAWSMskCluster_brokerNodes (1096.43s)
--- PASS: TestAccAWSMskCluster_tagsUpdate (1130.02s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       1130.671s
...
```

### CODE REVIEW

**Comment by KellerII** [comment]
Any idea on when this will be ready/merged?

File: website/docs/r/msk_cluster.html.markdown [hunk 1]
```diff
@@ -0,0 +1,131 @@
+---
+layout: "aws"
+page_title: "AWS: aws_msk_cluster"
+sidebar_current: "docs-aws-resource-msk-cluster"
+description: |-
+  Terraform resource for managing an AWS Managed Streaming for Kafka cluster
+---
+
+# aws_msk_cluster
```
Comment (bflad) [inline website/docs/r/msk_cluster.html.markdown]: Nit: We're now including `Resource: ` in resource titles 👍 

```suggestion
# Resource: aws_msk_cluster
```

File: website/docs/r/msk_cluster.html.markdown [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+Manages AWS Managed Streaming for Kafka cluster
+
```
Comment (bflad) [inline website/docs/r/msk_cluster.html.markdown]: _Maintainer Note:_ Adding the preview callout for this resource documentation

~> **NOTE:** This AWS service is in Preview and may change before General Availability release. Backwards compatibility is not guaranteed between Terraform AWS Provider releases.

File: website/aws.erb [hunk 3]
```diff
@@ -1876,6 +1876,16 @@
                     </ul>
                 </li>
 
+                <li>
+                    <a href="#">MSK Resources</a>
```
Comment (bflad) [inline website/aws.erb]: Nit: I'm thinking we might want to call out the full name here for folks searching for just "kafka", e.g.

```suggestion
                    <a href="#">Managed Streaming for Kafka (MSK) Resources</a>
```

File: aws/resource_aws_msk_cluster.go [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+			},
+			"encryption_info": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"encryption_at_rest_kms_id": {
```
Comment (bflad) [inline aws/resource_aws_msk_cluster.go]: Given the API always returns the ARN, we should just require inputting the ARN here to simplify attribute handling (removing the `DiffSuppressFunc`) and prevent potential "inconsistent plan" errors in Terraform 0.12 (the new "diffs didn't match during apply" error from Terraform 0.11).

```suggestion
						"encryption_at_rest_kms_key_arn": {
```

File: aws/resource_aws_msk_cluster.go [hunk 5]
```diff
@@ -0,0 +1,372 @@
...
+	return &schema.Resource{
+		Create: resourceAwsMskClusterCreate,
+		Read:   resourceAwsMskClusterRead,
+		Update: resourceAwsMskClusterUpdate,
+		Delete: resourceAwsMskClusterDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(45 * time.Minute),
```
Comment (bflad) [inline aws/resource_aws_msk_cluster.go]: It appears these customizable timeouts are not passed into the waiter functions (e.g. `waitForMskClusterCreation`) so they should be removed. 👍

**Comment by bflad** [comment]
This has been released in [version 2.12.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2120-may-24-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.

**Pull request merged**